flink-java/src/main/java/org/apache/flink/api/java/typeutils/runtime/EitherSerializer.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.typeutils.runtime;
+
+import java.io.IOException;
+
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+import org.apache.flink.api.java.typeutils.Either;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+
+/**
+ * A {@link TypeSerializer} for the {@ link Either} type of the Java class.
+ *
+ * @param <L> the Left value type
+ * @param <R> the Right value type
+ */
+public class EitherSerializer<L, R> extends TypeSerializer<Either<L, R>> {
+
+	private static final long serialVersionUID = 1L;
+
+	private final Class<Either<L, R>> typeClass;
+
+	private final TypeSerializer<L> leftSerializer;
+
+	private final TypeSerializer<R> rightSerializer;
+
+	public EitherSerializer(Class<Either<L, R>> typeClass, TypeSerializer<L> leftSerializer, TypeSerializer<R> rightSerializer) {
+		this.typeClass = typeClass;
+		this.leftSerializer = leftSerializer;
+		this.rightSerializer = rightSerializer;
+	}
+
+	@Override
+	public boolean isImmutableType() {
+		return true;
```

Comment:
I think this needs to be `leftSerializer.isImmutableType() & rightSerializer.isImmutableType()`.